lib/logflare/backends.ex

```diff
@@ -263,6 +282,53 @@ defmodule Logflare.Backends do
     |> Repo.delete()
   end
 
+  @doc """
+  Lists all sources associated with a backend.
+  """
+  @spec list_sources_for_backend(integer()) :: [Source.t()]
+  def list_sources_for_backend(backend_id) when is_integer(backend_id) do
+    case __MODULE__.Cache.get_backend(backend_id) do
+      %Backend{sources: [%Source{} | _] = sources} ->
+        sources
+
+      %Backend{} = backend ->
+        backend = Repo.preload(backend, :sources)
+        backend.sources
+
+      nil ->
+        []
+    end
+  end
+
+  @doc """
+  Syncs backend across all cluster nodes for v2 pipeline sources.
+
+  This ensures that when a backend's configuration changes (e.g., `default_ingest?` flag),
+  all running `SourceSup` instances are updated immediately without waiting for the
+  periodic `SourceSupWorker` check.
+  """
+  @spec sync_backend_across_cluster(integer()) :: :ok
+  def sync_backend_across_cluster(backend_id) when is_integer(backend_id) do
+    case __MODULE__.Cache.get_backend(backend_id) do
```

Comment:
likewise, doing a `Sources.list_sources` and working from fresh data would be much cleaner.

Also, anything in the Backends.* namespace is always for v2 sources, so we don't need to check for v2 pipeline.